### PR TITLE
[flutter_tools] remove more globals from `IOSDevices` and `IOSDevice`

### PR DESCRIPTION
**Pull request by christopherfujino**
## Description

Followup globals cleanup to https://github.com/flutter/flutter/pull/50772.

1. Removed a layer of indirection with `IOSDevices` instance methods `pollingGetDevices()` and `getDiagnostics()` merely being wrappers (that pass globals) around `IOSDevice` static methods `getAttachedDevices` and `getDiagnostics()` respectively. As these were the only invocations of the two static methods, I removed them and moved the (minimal) logic into the `IOSDevices` instance methods.
1. Removed globals from `IOSDevices` class.
1. Removed global logger from `IOSDevice` class.
1. Moved `iosWorkflow` context getter to `globals.dart`.

## Related Issues

Part of https://github.com/flutter/flutter/issues/47161

## Tests

Removed context from several tests.

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I signed the [CLA].
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] All existing and new tests are passing.
- [x] The analyzer (`flutter analyze --flutter-repo`) does not report any problems on my PR.
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Did any tests fail when you ran them? Please read [Handling breaking changes].

- [x] No, no existing tests failed, so this is *not* a breaking change.
- [ ] Yes, this is a breaking change. *If not, delete the remainder of this section.*
   - [ ] I wrote a design doc: https://flutter.dev/go/template *Replace this with a link to your design doc's short link*
   - [ ] I got input from the developer relations team, specifically from: *Replace with the names of who gave advice*
   - [ ] I wrote a migration guide: *Replace with a link to your migration guide*

### CODE REVIEW

File: packages/flutter_tools/test/general.shard/ios/devices_test.dart [hunk 1]
```diff
@@ -97,6 +97,7 @@ void main() {
         'device-123',
         artifacts: mockArtifacts,
         fileSystem: mockFileSystem,
+        logger: mockLogger,
```
Comment (jonahwilliams) [inline packages/flutter_tools/test/general.shard/ios/devices_test.dart]: If you're not specifically mocking some functionality of the logger for testing, use BufferLogger.test

Comment (christopherfujino) [reply packages/flutter_tools/test/general.shard/ios/devices_test.dart]: Yup, I'm resolving the merge conflict now and cleaning this up.

Comment (christopherfujino) [reply packages/flutter_tools/test/general.shard/ios/devices_test.dart]: resolved merge and removed all `mockLogger` instances, PTAL.

**Review by jonahwilliams — APPROVED** [review]
LGTM!

**Comment by christopherfujino** [comment]
I will not land this PR as is, and instead break it up into two PRs to make it a soft breaking change.

**Pull request closed without merging**